151
리스트에는 네 개의 정수가 저장돼 있다.

리스트 = [3, -20, -3, 44]
for문을 사용해서 리스트의 음수를 출력하라.

-20
-3

In [19]:
numlist=[3,-20,-3,44]
for i in numlist:
    if i<0:
        print(i)

-20
-3


152
for문을 사용해서 3의 배수만을 출력하라.

리스트 = [3, 100, 23, 44]

In [20]:
numlist=[3,100,23,44]
for i in numlist:
    if i%3==0:
        print(i)

3


153
리스트에서 20 보다 작은 3의 배수를 출력하라

리스트 = [13, 21, 12, 14, 30, 18]

In [21]:
numnum = [13, 21, 12, 14, 30, 18]
for i in numnum:
    if i<20 and i%3==0:
        print(i)

12
18


154
리스트에서 세 글자 이상의 문자를 화면에 출력하라

리스트 = ["I", "study", "python", "language", "!"]

In [22]:
words = ["I", "study", "python", "language", "!"]
for i in words:
    if len(i)>=3:
        print(i)

study
python
language


155 리스트에서 대문자만 화면에 출력하라.

리스트 = ["A", "b", "c", "D"]

In [25]:
alphabet =["A", "b", "c", "D"]
for i in alphabet:
    if i.isupper()==True:
        print(i)

A
D


156 리스트에서 소문자만 화면에 출력하라.

In [26]:
alphabet =["A", "b", "c", "D"]
for i in alphabet:
    if i.isupper()==False:
        print(i)

b
c


157
이름의 첫 글자를 대문자로 변경해서 출력하라.

리스트 = ['dog', 'cat', 'parrot']

In [27]:
animal = ['dog', 'cat', 'parrot']
for i in animal:
    capital=i[0];
    rest=i[1:];
    print(capital.upper()+rest)

Dog
Cat
Parrot


158 파일 이름이 저장된 리스트에서 확장자를 제거하고 파일 이름만 화면에 출력하라. (힌트: split() 메서드)

리스트 = ['hello.py', 'ex01.py', 'intro.hwp']

In [30]:
files = ['hello.py', 'ex01.py', 'intro.hwp']
for i in files:
    name=i.split(".")
    print(name[0])

hello
ex01
intro


159 파일 이름이 저장된 리스트에서 확장자가 .h인 파일 이름을 출력하라.

리스트 = ['intra.h', 'intra.c', 'define.h', 'run.py']        
 
intra.h
define.h

In [34]:
file= ['intra.h', 'intra.c', 'define.h', 'run.py']
for i in file:
    names=i.split(".")
    if names[1]=='h':
        print(i)

intra.h
define.h


In [ ]:
160
파일 이름이 저장된 리스트에서 확장자가 .h나 .c인 파일을 화면에 출력하라.

리스트 = ['intra.h', 'intra.c', 'define.h', 'run.py']

In [37]:
file= ['intra.h', 'intra.c', 'define.h', 'run.py']
for i in file:
    names=i.split(".")
    if names[1]=='h' or names[1]=='c':
        print(i)

intra.h
intra.c
define.h


161
for문과 range 구문을 사용해서 0~99까지 한 라인에 하나씩 순차적으로 출력하는 프로그램을 작성하라.

In [1]:
for i in range(100):
    print(i)

0
1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99


162
월드컵은 4년에 한 번 개최된다. range()를 사용하여 2002~2050년까지 중 월드컵이 개최되는 연도를 출력하라.

In [2]:
for i in range(2002,2051,4):
        print(i)

2002
2006
2010
2014
2018
2022
2026
2030
2034
2038
2042
2046
2050


163
1부터 30까지의 숫자 중 3의 배수를 출력하라.

In [3]:
for i in range(3,31,3):
    print(i)

3
6
9
12
15
18
21
24
27
30


165
for문을 사용해서 아래와 같이 출력하라.
0.0
0.1
0.2
0.3
0.4
0.5
...
0.9

In [4]:
for i in range(10):
    print(i/10)

0.0
0.1
0.2
0.3
0.4
0.5
0.6
0.7
0.8
0.9


166
구구단 3단을 출력하라.

3x1 = 3
3x2 = 6
3x3 = 9
3x4 = 12
3x5 = 15
3x6 = 18
3x7 = 21
3x8 = 24
3x9 = 27

In [7]:
for i in range(1,10):
    print("3x",i,"=",3*i)

3x 1 = 3
3x 2 = 6
3x 3 = 9
3x 4 = 12
3x 5 = 15
3x 6 = 18
3x 7 = 21
3x 8 = 24
3x 9 = 27


167
구구단 3단을 출력하라. 단 홀수 번째만 출력한다.

In [8]:
for i in range(1,10,2):
    print("3x",i,"=",3*i)

3x 1 = 3
3x 3 = 9
3x 5 = 15
3x 7 = 21
3x 9 = 27


168
1~10까지의 숫자에 대해 모두 더한 값을 출력하는 프로그램을 for 문을 사용하여 작성하라.

In [12]:
num=0;
for i in range(1,11):
    num+=i;
print(num)

55


169
1~10까지의 숫자 중 모든 홀수의 합을 출력하는 프로그램을 for 문을 사용하여 작성하라.

In [14]:
num=0;
for i in range(1,11,2):
    num+=i;
print(num)

25


170
1~10까지의 숫자를 모두 곱한 값을 출력하는 프로그램을 for 문을 사용하여 작성하라.

In [15]:
gob=1;
for i in range(1,11):
    gob=gob*i
print(gob)

3628800


171
아래와 같이 리스트의 데이터를 출력하라. 단, for문과 range문을 사용하라.

price_list = [32100, 32150, 32000, 32500]

32100
32150
32000
32500

In [38]:
price_list = [32100, 32150, 32000, 32500]
for i in range(4):
    print (price_list[i])

32100
32150
32000
32500


172
아래와 같이 리스트의 데이터를 출력하라. 단, for문과 range문을 사용하라.

price_list = [32100, 32150, 32000, 32500]           

 
0 32100
1 32150
2 32000
3 32500

In [39]:
for i in range(4):
    print(i, price_list[i])

0 32100
1 32150
2 32000
3 32500


아래와 같이 리스트의 데이터를 출력하라. 단, for문과 range문을 사용하라.

price_list = [32100, 32150, 32000, 32500]         
3 32100
2 32150
1 32000
0 32500

In [41]:
for i in range(4):
    print(3-i, price_list[i])

3 32100
2 32150
1 32000
0 32500


174
아래와 같이 리스트의 데이터를 출력하라. 단, for문과 range문을 사용하라.

price_list = [32100, 32150, 32000, 32500]              
100 32150     
110 32000      
120 32500       

In [42]:
for i in range(1,4):
    print(90+i*10,price_list[i])

100 32150
110 32000
120 32500


175
my_list를 아래와 같이 출력하라.

my_list = ["가", "나", "다", "라"]           
가 나      
나 다       
다 라      

In [43]:
my_list = ["가", "나", "다", "라"]
for i in range(0,3):
    print (my_list[i],my_list[i+1])

가 나
나 다
다 라


176
리스트를 아래와 같이 출력하라.

my_list = ["가", "나", "다", "라", "마"]       
가 나 다       
나 다 라       
다 라 마        

In [44]:
my_list = ["가", "나", "다", "라", "마"]
for i in range(0,3):
    print(my_list[i],my_list[i+1],my_list[i+2])

가 나 다
나 다 라
다 라 마


177 
반복문과 range 함수를 사용해서 my_list를 아래와 같이 출력하라.     

my_list = ["가", "나", "다", "라"]     

라 다      
다 나       
나 가       

In [45]:
my_list = ["가", "나", "다", "라"]
for i in range(0,3):
    print(my_list[3-i],my_list[2-i])

라 다
다 나
나 가


178  
리스트에는 네 개의 정수가 저장되어 있다. 각각의 데이터에 대해서 자신과 우측값과의 차분값을 화면에 출력하라.         

my_list = [100, 200, 400, 800]     
예를들어 100을 기준으로 우측에 위치한 200과의 차분 값를 화면에 출력하고, 200을 기준으로 우측에 위치한 400과의 차분값을 화면에 출력한다.   
이어서 400을 기준으로 우측에 위치한 800과의 차분값을 화면에 출력한다.         
       
100       
200        
400       

In [46]:
my_list = [100, 200, 400, 800]
for i in range(0,3):
    print(my_list[i+1]-my_list[i])

100
200
400


179
리스트에는 6일 간의 종가 데이터가 저장되어 있다. 종가 데이터의 3일 이동 평균을 계산하고 이를 화면에 출력하라.       
       
my_list = [100, 200, 400, 800, 1000, 1300]           
첫 번째 줄에는 100, 200, 400의 평균값이 출력된다.         
두 번째 줄에는 200, 400, 800의 평균값이 출력된다. 같은 방식으로 나머지 데이터의 평균을 출력한다.     

233.33333333333334      
466.6666666666667      
733.3333333333334      
1033.3333333333333     

In [48]:
my_list = [100, 200, 400, 800, 1000, 1300]
for i in range(0,4):
    print((my_list[i]+my_list[i+1]+my_list[i+2])/3)

233.33333333333334
466.6666666666667
733.3333333333334
1033.3333333333333


180        
리스트에 5일간의 저가, 고가 정보가 저장돼 있다. 고가와 저가의 차를 변동폭이라고 정의할 때, low, high 두 개의 리스트를 사용해서 5일간의 변동폭을 volatility 리스트에 저장하라.         

low_prices  = [100, 200, 400, 800, 1000]         
high_prices = [150, 300, 430, 880, 1000]    


In [52]:
low_prices = [100, 200, 400, 800, 1000]
high_prices = [150, 300, 430, 880, 1000]

volatility=[]
for i in range(5):
    volatility.append(high_prices[i]-low_prices[i])
    
print(volatility)    

[50, 100, 30, 80, 0]


181
아래 표에서 하나의 행을 하나의 리스트로, 총 3개의 리스트를 갖는 이차원 리스트 apart를 정의하라.
 
101호	102호      
201호	202호   
301호	302호      

In [53]:
apart = [ ["101호", "102호"], ["201호", "202호"], ["301호", "302호"] ]

182
아래 표에서 하나의 열을 하나의 리스트로, 총 2개의 리스트를 갖는 이차원 리스트 stock을 정의하라.

시가	종가       
100	80       
200	210       
300	330       

In [54]:
stock = [ ["시가", 100, 200, 300], ["종가", 80, 210, 330] ]

183
아래 표를 stock 이름의 딕셔너리로 표현하라.시가를 key로 저장하고, 나머지 같은 열의 데이터를 리스트로 저장해서 value로 저장한다. 종가 역시 key로 저장하고 나머지 같은 열의 데이터를 리스트로 저장해서 value로 저장한다.

시가	종가    
100	80         
200	210    
300	330     

In [55]:
stock = {"시가": [100, 200, 300], "종가": [80, 210, 330] }

185
리스트에 저장된 데이터를 아래와 같이 출력하라.

apart = [ [101, 102], [201, 202], [301, 302] ]           
101 호         
102 호       
201 호         
202 호        
301 호       
302 호        

In [57]:
apart = [ [101, 102], [201, 202], [301, 302] ]
for i in apart:
    for j in i:
        print(j,"호")

101 호
102 호
201 호
202 호
301 호
302 호


186
리스트에 저장된 데이터를 아래와 같이 출력하라.             
 
apart = [ [101, 102], [201, 202], [301, 302] ]         
301 호       
302 호        
201 호       
202 호      
101 호      
102 호      

In [60]:
for i in apart[::-1]:
    for j in i:
        print(j,"호")

301 호
302 호
201 호
202 호
101 호
102 호


187
리스트에 저장된 데이터를 아래와 같이 출력하라.          

apart = [ [101, 102], [201, 202], [301, 302] ]        
302 호       
301 호        
202 호       
201 호      
102 호        
101 호      

In [61]:
for i in apart[::-1]:
    for j in i[::-1]:
        print(j,"호")

302 호
301 호
202 호
201 호
102 호
101 호


188
리스트에 저장된 데이터를 아래와 같이 출력하라.

apart = [ [101, 102], [201, 202], [301, 302] ]       

       
101 호
-----
102 호
-----
201 호
-----
202 호
-----
301 호
-----
302 호
-----

In [63]:
for i in apart:
    for j in  i:
        print(j, "호")
        print("-" * 5)

101 호
-----
102 호
-----
201 호
-----
202 호
-----
301 호
-----
302 호
-----


189
리스트에 저장된 데이터를 아래와 같이 출력하라.

apart = [ [101, 102], [201, 202], [301, 302] ]


101 호          
102 호
-----       
201 호        
202 호      
-----       
301 호       
302 호
-----

In [64]:
for i in apart:
    for j in i:
        print(j, "호")
    print("-----")

101 호
102 호
-----
201 호
202 호
-----
301 호
302 호
-----


101 호       
102 호       
201 호      
202 호      
301 호     
302 호      
-----


In [66]:
for i in apart:
    for j in i:
        print(j, "호")
print("-" * 5)

101 호
102 호
201 호
202 호
301 호
302 호
-----


191
data에는 매수한 종목들의 OHLC (open/high/low/close) 가격 정보가 바인딩 되어있다.                   

data = [
    [ 2000,  3050,  2050,  1980],
    [ 7500,  2050,  2050,  1980],
    [15450, 15050, 15550, 14900]
]

수수료를 0.014 %로 가정할 때, 각 가격에 수수료를 포함한 가격을 한라인에 하나씩 출력하라.          

2000.28       
3050.427           
2050.2870000000003        
...        

In [70]:
data = [
    [ 2000,  3050,  2050,  1980],
    [ 7500,  2050,  2050,  1980],
    [15450, 15050, 15550, 14900]
]

for i in data:
    for j in i:
        print(float(j*1.00014))

2000.28
3050.427
2050.2870000000003
1980.2772
7501.05
2050.2870000000003
2050.2870000000003
1980.2772
15452.163
15052.107
15552.177
14902.086000000001


192
191번의 출력 결과에 행단위로 "----" 구분자를 추가하라.

In [71]:
for i in data:
    for j in i:
        print(float(j*1.00014))
    print("----")

2000.28
3050.427
2050.2870000000003
1980.2772
----
7501.05
2050.2870000000003
2050.2870000000003
1980.2772
----
15452.163
15052.107
15552.177
14902.086000000001
----


193
192 번 문제의 결괏값을 result 이름의 리스트에 1차원 배열로 저장하라.



In [72]:
result=[]

for i in data:
    for j in i:
        result.append(float(j*1.00014))

print(result)

[2000.28, 3050.427, 2050.2870000000003, 1980.2772, 7501.05, 2050.2870000000003, 2050.2870000000003, 1980.2772, 15452.163, 15052.107, 15552.177, 14902.086000000001]


194
191번 문제의 결괏값을 result 이름의 리스트에 2차원 배열로 저장하라. 저장 포맷은 아래와 같다. 각 행에 대한 데이터끼리 리스트에 저장되어야 한다.

[ [2000.28, 3050.427, 2050.2870000000003, 1980.2772],
 [7501.05, 2050.2870000000003, 2050.2870000000003, 1980.2772],
 [15452.163, 15052.107, 15552.177, 14902.086000000001]]

In [73]:
result=[]

for i in data:
    result2 = []
    for j in i:
        result2.append(j * 1.00014)
    result.append(result2)
print(result)

[[2000.28, 3050.427, 2050.2870000000003, 1980.2772], [7501.05, 2050.2870000000003, 2050.2870000000003, 1980.2772], [15452.163, 15052.107, 15552.177, 14902.086000000001]]


195 ohlc 리스트에는 시가(open), 고가 (high), 저가 (low) , 종가(close)가 날짜별로 저장돼 있다. 화면에 종가데이터를 출력하라.         
       
ohlc = [["open", "high", "low", "close"],
        [100, 110, 70, 100],
        [200, 210, 180, 190],
        [300, 310, 300, 310]]

In [75]:
ohlc = [["open", "high", "low", "close"],
        [100, 110, 70, 100],
        [200, 210, 180, 190],
        [300, 310, 300, 310]]

for i in ohlc[1:]:
    print(i[3])

100
190
310


196 종가가 150원보다 큰경우에만 종가를 출력하라.

In [76]:
for i in ohlc[1:]:
    if i[3]>150:
        print(i[3])

190
310


197  종가가 시가 보다 크거나 같은 경우에만 종가를 출력하라.

In [78]:
for i in ohlc[1:]:
    if i[3]>=i[0]:
        print(i[3])

100
310


198 고가와 저가의 차이를 변동폭으로 정의할 때 변동폭을 volatility 이름의 리스트에 저장하라.

In [79]:
volatility=[]


for i in ohlc[1:]:
    volatility.append(i[1]-i[2])
    
print(volatility)

[40, 30, 10]


199
리스트에는 3일 간의 ohlc 데이터가 저장돼 있다. 종가가 시가보다 높은 날의 변동성 (고가 - 저가)을 화면에 출력하라.

In [80]:
for i in ohlc[1:]:
    if i[3] > i[0]:
        print(i[1]-i[2])

10


시가에 매수해서 종가에 매도 했을 경우 총 수익금을 계산하라.

In [81]:
money = 0
for row in ohlc[1:]:
    money += (row[3] - row[0])

In [82]:
print(money)

0
